src/Console/Command/PostExporter.php

```diff
@@ -59,95 +41,32 @@ class PostExporter extends Command
-
-		// Construct a Search Data object to hold the search info
-		$data = $this->data->get('search');
-
-		// Get CLI params
-		$limit = $this->option('limit');
-		$offset = $this->option('offset');
-		$job_id = $this->argument('job');
-		$add_header = $this->option('include-header');
-
-		// At the moment there is only CSV format
-		$format = 'csv';
-
-		// Set the baseline filter parameters
-		$filters = [
-			'limit' => $limit,
-			'offset' => $offset,
-			'exporter' => true
-		];
-
-
-		// Load the export job
-		$job = $this->exportJobRepository->get($job_id);
-
-		$this->session->setUser($job->user_id);
-		// Merge the export job filters with the base filters
-		if ($job->filters) {
-			$filters = array_merge($filters, $job->filters);
-		}
-
-		// Set the fields that should be included if set
-		if ($job->fields) {
-			$data->include_attributes = $job->fields;
-		}
-
-		foreach ($filters as $key => $filter) {
-			$data->$key = $filter;
+		if (!$this->usecase) {
+			// @todo inject
+			$this->usecase = service('factory.usecase')
+				->get('posts_export', 'export');
 		}
+		return $this->usecase;
+	}
 
-		$this->postExportRepository->setSearchParams($data);
-		$posts = $this->postExportRepository->getSearchResults();
-		$this->formatter->setAddHeader($add_header);
-		//fixme add post_date
-		$form_ids = $this->postExportRepository->getFormIdsForHeaders();
-		$attributes = $this->formAttributeRepository->getByForms($form_ids, $data->include_attributes);
-
-		$keyAttributes = [];
-		foreach ($attributes as $key => $item) {
-			$keyAttributes[$item['key']] = $item;
-		}
-
-		// // ... remove any entities that cannot be seen
-		foreach ($posts as $idx => $post) {
-			// Retrieved Attribute Labels for Entity's values
-			$post = $this->postExportRepository->retrieveMetaData($post->asArray(), $keyAttributes);
-			$posts[$idx] = $post;
-		}
-		
-		if (empty($job->header_row)) {
-			$job->setState(['header_row' => $attributes]);
-			$this->exportJobRepository->update($job);
-		}
-
-		$header_row = $this->formatter->createHeading($job->header_row);
-		
-		$this->formatter->setHeading($header_row);
-		$formatter = $this->formatter;
-		/**
-		 * KeyAttributes is sent instead of the header row because it contains
-		 * the attributes with the corresponding features (type, priority) that
-		 * we need for manipulating the data
-		 */
-		$file = $formatter($posts, $keyAttributes);
-
-		$response = [
-			[
-				'file' => $file->file,
-			]
+	public function handle()
+	{
+		// set CLI params to be the payload for the usecase
+		$payload = [
+			'job_id' => $this->argument('job'),
+			'limit' => $this->option('limit'),
+			'offset' => $this->option('offset'),
+			'add_header' => $this->option('include-header'),
 		];
 
-		$this->line(json_encode($response));
+		// Get the usecase and pass in authorizer, payload and transformer
+		$file  = $this->getUsecase()
+			->setPayload($payload)
+			->setAuthorizer(service('authorizer.export_job'))
```

Comment:
I think setAuthorizer and setFormatter should all happen in `getUsecase` or update the DI  setup so they're set correctly by the usecase factory first